bug: `moduleResolution` is not respected

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

3.2.0

### Current Behavior

I have a library that is used by my component and the library has sub modules that is using the new [`exports`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing) syntax in `package.json`. My component is importing it something like this
```ts
import { foo } from 'my-lib/foo`;
```

To get it to work I have set `moduleResolution` in `tsconfig.json` to `node16`/`nodenext`, but when I build my project I get an error saying 
```
[ ERROR ]  TypeScript: ./src/components/my-component/my-component.tsx:3:21
           Cannot find module 'my-lib/foo' or its corresponding type declarations.

      L2:  import { myLib } from 'my-lib';
      L3:  import { foo } from 'my-lib/foo';

[28:31.4]  build failed, watching for changes... in 3 ms

``` 

VSCode can find the module when I set `moduleResolution` to `node16`, and I can run the TypeScript compiler manually and it will compile, but it seems Stencil is not respecting the `moduleResolution` setting in `tsconfig.json`

### Expected Behavior

No build errors. Stencil should use the `moduleResolution` specified in `tsconfig.json`

### System Info

```shell
System: node 16.10.0
    Platform: linux (5.19.0-38-generic)
   CPU Model: 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 cpus)
    Compiler: /home/foo/node_modules/@stencil/core/compiler/stencil.js
       Build: 1678807683
     Stencil: 3.2.0 🌷
  TypeScript: 4.9.5
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.16.6
```


### Steps to Reproduce

Build the included Stencil project

### Code Reproduction URL

https://github.com/jgroth/stencil-module-resolution

### Additional Information

_No response_

--- a/src/compiler/transpile/ts-config.ts
+++ b/src/compiler/transpile/ts-config.ts
@@ -26,12 +26,12 @@ export const getTsOptionsToExtend = (config: d.ValidatedConfig): ts.CompilerOpti
     // if the `DIST_TYPES` output target is present then we'd like to emit
     // declaration files
     declaration: config.outputTargets.some(isOutputTargetDistTypes),
-    module: config.tsCompilerOptions.module || ts.ModuleKind.ESNext,
+    module: config.tsCompilerOptions?.module || ts.ModuleKind.ESNext,
     moduleResolution:
-      config.tsCompilerOptions.moduleResolution === ts.ModuleResolutionKind.Bundler
+      config.tsCompilerOptions?.moduleResolution === ts.ModuleResolutionKind.Bundler
         ? ts.ModuleResolutionKind.Bundler
         : ts.ModuleResolutionKind.NodeJs,
-    noEmitOnError: config.tsCompilerOptions.noEmitOnError || false,
+    noEmitOnError: config.tsCompilerOptions?.noEmitOnError || false,
     outDir: config.cacheDir || config.sys.tmpDirSync(),
     sourceMap: config.sourceMap,
     inlineSources: config.sourceMap,

